### PR TITLE
Category for Thing types

### LINKED ISSUE #568: Add Category to thing-type
migrated from Bugzilla [#465478](https://bugs.eclipse.org/bugs/show_bug.cgi?id=465478)
status UNCONFIRMED severity _enhancement_ in component _Core_ for _---_
Reported in version _unspecified_ on platform _PC_
Assigned to: Project Inbox

On 2015-04-25 07:19:04 -0400, Chris Jackson wrote:

> I think it would be useful to add category to the ThingType object. 
> 
> Currently Category a member of the channel and item, and therefore is effectively added to a Thing once it's created, but I think it would make sense to have it available in the ThingType as well to allow users to sort/find/display information.

On 2015-04-25 08:08:33 -0400, Dennis Nobel wrote:

> All the categories we defined so far are "functional categories". The problem with things or devices is, that the diversity is huge. Functions are the atomic buildings blocks of devices. When trying to categorize devices you end up creating an ontology, where a lot of people failed so far. Because of this we decided to not have a category for things.
> 
> For sure for some simple devices it seems to be straight forward. E.g. a thermostat could be a category for a device that supports temperature regulation. But what is the category of a more complex devices with multiple functions. Kai just me showed me the new Fibaro motion sensor (http://www.fibaro.com/de/the-fibaro-system/motion-sensor). Besides motion it also measures temperature and other values. So what should be the category of this device? "motion sensor"? "motion sensor with temperature sensor"? "device with multiple sensors"? I hope the problem is clear.
> 
> A lot of people and organizations try to create ontologies at the moment. Before we also try to create a new one, I would propose to wait until someone else comes up with a good approach (if this is possible at all - IÂ´m not yet convinced that this is possible).

On 2015-04-27 09:43:28 -0400, Chris Jackson wrote:

> I agree - there's a log of diversity in sensors these days, but to answer your question about the Fibaro FGMS - it's a motion sensor. Yes, it has secondary sensors that can be enabled separately (disabled by default), but it's sold as a motion sensor.
> 
> In any case, I understand what you mean, but I also think that some sort of category (or maybe we call it something else) is useful. We already have the ability to put a category into a thing (ie in the thing-item), so it seems a little strange that it can't also be attributed to the thing-type?!?
> 
> From a UI perspective, I think it's useful to have some field attributed to a device that allows searching. Maybe it can be done indirectly through the channel categories, but there's no way to say which is the 'main' channel and what are secondary. I guess that's another solution is to add some sort of 'channel context' so that we know if the channel is a primary or secondary (or third-ary) function of the device?

### PR DESCRIPTION
**Pull request by triller-telekom**
Added "category" to thing types and group channel types

Fixes #568

### CODE REVIEW

**Comment by triller-telekom** [comment]
@kaikreuzer Can you please restart Travis?

**Comment by kaikreuzer** [comment]
sure

**Comment by sjka** [comment]
This time eclipse.org had hick-ups. Hilarious.
I restarted the build again.

**Comment by maggu2810** [comment]
There is a defined set of categories for channels in the documentation.

Can we introduce some initial thing types with the introduction of the thing type support (not only a "TODO: to be specified soon")?
In the linked issue you talk about channel groups type could be similar to "sub-devices". So, are the possible categories of a channel group type equal to the ones for thing types, a sub set or a concat between channel type categories and thing type categories?

Can you give me some more information about the whole topic?

**Comment by triller-telekom** [comment]
@maggu2810 This is a very old issue and the idea was to provide some "search-able" information about `things`. If a user has a lot of devices which will be discovered as `things` in his `Inbox` it would be nice to have a search function. The user might be interested in all `Alarm things` for example.

Since `channels` do have a `category` already once could indirectly derive one for a `thing`, but the problem is that a `thing` can have multiple `channels`. So this new `thing category` would be a description for the "primary purpose" of a `thing`.

As a starting point I would take the `channel categories` as `thing categories` which we have specified here: https://github.com/eclipse/smarthome/blob/master/docs/documentation/development/bindings/thing-definition.md#channel-categories

But I wanted to leave this open for a discussion, hence the "Todo" ;) 

Regarding the mentioned "sub-devices from @kaikreuzer I would say that `channel group types`are the same as `thing types` as a starting point. Maybe later one could replace them by a proper ontology?

**Comment by kaikreuzer** [comment]
> Can we introduce some initial thing types with the introduction of the thing type support (not only a "TODO: to be specified soon")?

I agree - this should be the main deliverable of this issue: A good list of categories for things / channelgroups and an extended one for channels.

> the idea was to provide some "search-able" information about things

Imho the main purpose will be to show an adequate icon next to a Thing (in the thing list or the inbox likewise).

> The user might be interested in all Alarm things for example.
> this new thing category would be a description for the "primary purpose" 

Don't mix up categorisation/taxonomy with the ontology discussion. The categories are not about the purpose, but about what kind of device it is (and thus being able to show a nice icon for it). So there e.g. should be a category for a "smart plug", which does not tell much about its purpose.

**Comment by triller-telekom** [comment]
I have just created an initial list of thing types:
https://github.com/eclipse/smarthome/pull/3510/commits/5725a596b75fc6b81bdbc8ea4f83f64c3151a38d#diff-f9de0c4b78c720e72b52a8a928b456d1R11

I tried to group as many devices as possible under each category. My intention behind this is to lower the amount of work for people who want to provide a custom iconset. However if you think that each `thing` that I have listed in the description of a category should have its own category we can still do that.

What do you guys think about it? Please have a look at this initial list.

**Comment by sjka** [comment]
IMHO some of the Thing categories focus on functional categories too much already. You can already see this by looking at your descriptions: Many of them define the categories by explaining what they "provide". The main benefit of having such categories is that UIs can render them generically. So rather think about how the devices may look like instead of what you can do with them. 

Some examples:
* You have _Door_, _Window_ and _BinarySensor_ - in the end this is all the same: a reed contact, most of the time inside a small rectangle case, which you can glue somewhere (and the Thing itself does not care if it is a window or a door)
* Television: nice functional category. However, how would you draw _one_ picture for TV, recorder, projector? I'd rather distinguish _screen_, _hifi_, _projector_
* _Food_: Again, a _mixer_ looks fundamentally different than a _"white" kitchen appliance_ like e.g. a fridge
* ...

Wouln't it rather make sense to thing along the lines of their visual appearance, e.g. _DIN rail_ device, _(plugable) socket_ device, _integrated home appliance_, _"white" home appliance_ and the like?

**Comment by triller-telekom** [comment]
To simplify the discussion I removed `category` on the `group channel types` from this PR an created  a separate one to discuss those, see #3629

**Comment by triller-telekom** [comment]
rebased against current master

File: bundles/core/org.eclipse.smarthome.core.thing.xml.test/src/test/resources/test-bundle-pool/ThingTypesTest.bundle/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -10,6 +10,8 @@
 		<description>The hue Bridge represents the Philips hue bridge.
 		</description>
 
+		<category>Light</category>
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing.xml.test/src/test/resources/test-bundle-pool/ThingTypesTest.bundle/ESH-INF/thing/thing-types.xml]: This should rather "Gateway" (or what category name we end up for it), shouldn't it?

File: bundles/core/org.eclipse.smarthome.core.thing.xml.test/src/test/resources/test-bundle-pool/ThingTypesTest.bundle/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -41,6 +43,8 @@
 		<label>HUE Lamp</label>
 		<description>My own great HUE Lamp.</description>
 
+		<category>Light</category>
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing.xml.test/src/test/resources/test-bundle-pool/ThingTypesTest.bundle/ESH-INF/thing/thing-types.xml]: Probably better a "Lightbulb"

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ThingTypeI18nLocalizationService.java [hunk 3]
```diff
@@ -53,21 +53,10 @@ public ThingType createLocalizedThingType(Bundle bundle, ThingType thingType, Lo
                 thingType.getChannelDefinitions().size());
 
         for (final ChannelDefinition channelDefinition : thingType.getChannelDefinitions()) {
-            String channelLabel = this.thingTypeI18nUtil.getChannelLabel(bundle, thingType.getUID(), channelDefinition,
-                    channelDefinition.getLabel(), locale);
-            String channelDescription = this.thingTypeI18nUtil.getChannelDescription(bundle, thingType.getUID(),
-                    channelDefinition, channelDefinition.getDescription(), locale);
-            if (channelLabel == null || channelDescription == null) {
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/i18n/ThingTypeI18nLocalizationService.java]: Is it really correct to remove all those null checks here from the code?

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/BridgeType.java [hunk 4]
```diff
@@ -86,6 +86,8 @@ public BridgeType(ThingTypeUID uid, List<String> supportedBridgeTypeUIDs, String
      * @param description the human readable description for the according type
      *            (could be null or empty)
      *
+     * @param category provides information about the thing for filtering
+     *
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/BridgeType.java]: I see that you didn't start with it, but could you remove all those silly empty lines between the params?

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/BridgeType.java [hunk 5]
```diff
@@ -124,6 +126,8 @@ public BridgeType(ThingTypeUID uid, List<String> supportedBridgeTypeUIDs, String
...
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/BridgeType.java]: same here

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/ThingType.java [hunk 6]
```diff
@@ -154,12 +158,14 @@ public ThingType(ThingTypeUID uid, List<String> supportedBridgeTypeUIDs, String
...
     public ThingType(ThingTypeUID uid, List<String> supportedBridgeTypeUIDs, String label, String description,
-            boolean listed, String representationProperty, List<ChannelDefinition> channelDefinitions,
+            String category, boolean listed, String representationProperty, List<ChannelDefinition> channelDefinitions,
             List<ChannelGroupDefinition> channelGroupDefinitions, Map<String, String> properties,
             URI configDescriptionURI) throws IllegalArgumentException {
 
         super(uid, label, description);
 
+        this.category = category;
+
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/ThingType.java]: could we remove all those empty lines?

File: docs/documentation/development/bindings/categories.md [hunk 7]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
 
 | Category        | Description                                          |
 |-----------------|------------------------------------------------------|
-
+| Aircondition    | Air condition devices incl. Fans |
+| BinarySensor    | General purpose sensor that returns a boolean value, i.e. a contact that reports open/close on a door or window |
+| Bridge          | Bridges/Gateway need to access other devices like used by Philips Hue for example |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: Might ["NetworkAppliance"](https://www.google.de/search?q=network+appliance&tbm=isch) be a good, more general category?

File: docs/documentation/development/bindings/categories.md [hunk 8]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
 
 | Category        | Description                                          |
 |-----------------|------------------------------------------------------|
-
+| Aircondition    | Air condition devices incl. Fans |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: Aircondition might be a bit specific, thinking that combined HVAC appliances are very wide-spread (especially in the US) - so HVAC might be a better choice.

File: docs/documentation/development/bindings/categories.md [hunk 9]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| BinarySensor    | General purpose sensor that returns a boolean value, i.e. a contact that reports open/close on a door or window |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: too functional

File: docs/documentation/development/bindings/categories.md [hunk 10]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Bridge          | Bridges/Gateway need to access other devices like used by Philips Hue for example |
+| Boiler          | Devices that heat up water |
+| Camera          | All kinds of cameras |
+| Car             | Smart Cars |
+| Cistern         | Cistern |
+| Clock           | Devices that provide the current time |
+| Coffee          | Coffee machines |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: -> CoffeeMachine

File: docs/documentation/development/bindings/categories.md [hunk 11]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Fridge          | Fridges, Freezers |
+| Furniture       | Smart furniture |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: imho too broad

File: docs/documentation/development/bindings/categories.md [hunk 12]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| GarageDoor      | Garage doors |
+| GeneralSensor   | General purpose sensor that does not fit in any category |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: Why not simply "Sensor"?

File: docs/documentation/development/bindings/categories.md [hunk 13]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Heating         | Devices that deal with heating |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: too broad / too functional. Maybe "Radiator"? But they are usually not connected anyhow. What we often have are ["RadiatorControls"](https://www.google.de/search?q=Radiator+Controls&tbm=isch).

File: docs/documentation/development/bindings/categories.md [hunk 14]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Hifi            | Devices related to TV/audio, such as satelite or cable receivers, recorders, etc.
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: Hifi does not seem an appropriate word, don't have any other at hand right now, though...

File: docs/documentation/development/bindings/categories.md [hunk 15]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| HomeApplicance  | Devices that look like Waschingmachines, Dishwashers, Dryers, etc. |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: This is far too broad - a home appliance is almost everything.
As we have "Fridge" as a separate category, we should probably do the same for the rest here -or we combine it as "white good".

File: docs/documentation/development/bindings/categories.md [hunk 16]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Light           | Devices that illuminate something, such as bulbs, etc. |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: too functional. We should have a "LightBulb" instead.

File: docs/documentation/development/bindings/categories.md [hunk 17]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Lock            | Devices whose primary pupose is locking something |
+| Microphone      | Devices to record sound |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: Do we have smart microphones anywhere? I don't really see the need.

File: docs/documentation/development/bindings/categories.md [hunk 18]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Mixer           | Kitchen devices for making cake |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: Do we have any such device? I don't see the need.

File: docs/documentation/development/bindings/categories.md [hunk 19]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| MotionSensor    | Motion sensors/detectors |
+| MultiSwitch     | Device which controls MULTIPLE states (else see Switch) of something |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: too functional

File: docs/documentation/development/bindings/categories.md [hunk 20]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Oven            | Ovens, microwaves, etc. |
+| Phone           | Anything related to telephony, such as mobile phones, landline phones, etc. |
+| PowerSocketDevice | Small devices to be plugged into a power socket in a wall which stick there |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: Is that the most common name for it?

File: docs/documentation/development/bindings/categories.md [hunk 21]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Heating         | Devices that deal with heating |
+| Hifi            | Devices related to TV/audio, such as satelite or cable receivers, recorders, etc.
+| HomeApplicance  | Devices that look like Waschingmachines, Dishwashers, Dryers, etc. |
+| Light           | Devices that illuminate something, such as bulbs, etc. |
+| Lock            | Devices whose primary pupose is locking something |
+| Microphone      | Devices to record sound |
+| Mixer           | Kitchen devices for making cake |
+| MotionSensor    | Motion sensors/detectors |
+| MultiSwitch     | Device which controls MULTIPLE states (else see Switch) of something |
+| Oven            | Ovens, microwaves, etc. |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: Maybe rather split oven and microwaves? They look pretty different...

File: docs/documentation/development/bindings/categories.md [hunk 22]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Phone           | Anything related to telephony, such as mobile phones, landline phones, etc. |
+| PowerSocketDevice | Small devices to be plugged into a power socket in a wall which stick there |
+| Projector       | Devices that project a picture somewhere |
+| Pump            | Water pumps |
+| Radiator        | Radiators that heat up rooms |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: see above

File: docs/documentation/development/bindings/categories.md [hunk 23]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Robots          | Vacuum cleaners such as cleaning robots, lawn mowers, etc. |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: Might also be a walking humanoid. Difficult to have such a general category. How do others do it?

File: docs/documentation/development/bindings/categories.md [hunk 24]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| RollerShutter   | Roller shutters |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: Maybe "Shade" would be more generic?

File: docs/documentation/development/bindings/categories.md [hunk 25]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Screen          | Devices that are able to show a picture |
+| Siren           | Siren used by Alarm systems |
+| Softener        | Water softeners |
+| SolarPlant      | Devices related to solar |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: too functional - I would have no idea what is behind here.

File: docs/documentation/development/bindings/categories.md [hunk 26]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Smoke           | Smoke detectors |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: -> SmokeDetector

File: docs/documentation/development/bindings/categories.md [hunk 27]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Speakers        | Devices that are able to play sounds |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: -> Speaker

File: docs/documentation/development/bindings/categories.md [hunk 28]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| (Wall)Switch    | Any device that controls a BINARY status (else see MultiSwitch) of something, for ex. a light switch |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: -> WallSwitch

File: docs/documentation/development/bindings/categories.md [hunk 29]
```diff
@@ -14,7 +14,44 @@ The thing type definition allows to specify a category. User interfaces can pars
...
+| Weather         | Rain sensors, Humidity sensors, Pressure sensors, wind sensors, etc. |
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/categories.md]: too functional. Maybe "WeatherStation" would be ok.

File: docs/documentation/development/bindings/thing-definition.md [hunk 30]
```diff
@@ -21,6 +21,7 @@ Things represent devices or services that can be individually added to, configur
         </supported-bridge-type-refs>
         <label>Sample Thing</label>
         <description>Some sample description</description>
+        <category>Light</category>
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/thing-definition.md]: adapt to something existing

File: docs/documentation/development/bindings/thing-definition.md [hunk 31]
```diff
@@ -212,7 +221,7 @@ The channel group type is defined on the same level as the thing types and chann
 </channel-group-type>
 ```
 
-When a thing will be created for a thing type with channel groups, the channel UID will contain the group ID in the last segment divided by a hash (#). If an Item should be linked to a channel within a group, the channel UID would be `binding:multiChannelSwitchActor:myDevice:switchActor1#switch` for the XML example before.
+When a thing will be created for a thing type with channel groups, the channel UID will contain the group ID in the last segment divided by a hash (#). If an Item should be linked to a channel within a group, the channel UID would be `binding:multiChannelSwitchActor:myDevice:switchActor1#switch` for the XML example before. Details about the category can be found in our [categories overview](categories.html).
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/thing-definition.md]: Please add a new line for every sentence.

File: docs/documentation/development/bindings/xml-reference.md [hunk 32]
```diff
@@ -446,6 +448,7 @@ Bridge and *Thing* descriptions must be placed as XML file(s) (with the ending `
   <tr><td>bridge-type-ref.id</td><td>The identifier of a bridge this bridge/<i>Thing</i> can connect to (mandatory).</td></tr>
   <tr><td>label</td><td>A human-readable label for the bridge/<i>Thing</i> (mandatory).</td></tr>
   <tr><td>description</td><td>A human-readable description for the bridge/<i>Thing</i> (optional).</td></tr>
+  <tr><td>category</td><td>Category this bridge/<i>Thing</i> belongs to to be used for filtering e.g. <code>Light</code> (optional).</td></tr>
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/xml-reference.md]: remove the example and the "filtering" use - maybe rather link to the categories from here.

File: extensions/ui/iconset/org.eclipse.smarthome.ui.iconset.classic/README.md [hunk 33]
```diff
@@ -3,1521 +3,83 @@
...
-  </figure>
-  <figure style="width: 128px; display: inline-block; text-align: center; font-size: 0.8em; margin: 16px 8px;">
-    <img src="icons/rain.png" alt="rain" title="rain">
-    <figcaption>rain</figcaption>
-  </figure>
-  <figure style="width: 128px; display: inline-block; text-align: center; font-size: 0.8em; margin: 16px 8px;">
-    <img src="icons/receiver.png" alt="receiver" title="receiver">
-    <figcaption>receiver</figcaption>
-  </figure>
-  <figure style="width: 128px; display: inline-block; text-align: center; font-size: 0.8
```
Comment (kaikreuzer) [inline extensions/ui/iconset/org.eclipse.smarthome.ui.iconset.classic/README.md]: You will have to invert the logic: You must iterate over all existing icons and check whether they belong to one of the categories. Currently, you are missing all the icons for specific states, i.e. while "light" is here, "light-on" and "light-off" are missing.

Comment (kaikreuzer) [inline extensions/ui/iconset/org.eclipse.smarthome.ui.iconset.classic/README.md]: I would prefer a less prominent representation - e.g. just an empty frame.
You also should not have to add a real icon to the set for that, so please remove the empty.png again. There should be other ways to render a placeholder here.

Comment (kaikreuzer) [inline extensions/ui/iconset/org.eclipse.smarthome.ui.iconset.classic/README.md]: All icons from the channel categories are still contained in that list and not filtered out as expected.

File: docs/documentation/development/bindings/xml-reference.md [hunk 34]
```diff
@@ -446,6 +448,7 @@ Bridge and *Thing* descriptions must be placed as XML file(s) (with the ending `
   <tr><td>bridge-type-ref.id</td><td>The identifier of a bridge this bridge/<i>Thing</i> can connect to (mandatory).</td></tr>
   <tr><td>label</td><td>A human-readable label for the bridge/<i>Thing</i> (mandatory).</td></tr>
   <tr><td>description</td><td>A human-readable description for the bridge/<i>Thing</i> (optional).</td></tr>
+  <tr><td>category</td><td>Category this bridge/<i>Thing</i> belongs to to, see [categories](categories.html) (optional).</td></tr>
```
Comment (kaikreuzer) [inline docs/documentation/development/bindings/xml-reference.md]: typo: to to

**Comment by triller-telekom** [comment]
Careful, this PR is API breaking because we renamed some icon files, namely:
```
projector_benq.png => projector.png
projector_benq.svg => projector.svg
television-off.png => screen-off.png
television-off.svg => screen-off.svg
television-on.png => screen-on.png
television-on.svg => screen-on.svg
television.png => screen.png
television.svg => screen.svg
```

File: docs/documentation/development/bindings/thing-definition.md [hunk 35]
```diff
@@ -35,6 +36,10 @@ Bindings may optionally set the listing of a thing type. By doing do, they indic
 
 Thing types are listed by default, unless specified otherwise. Hiding thing types potentially makes sense if they are deprecated and should not be used anymore. Also, this can be useful if users should not be bothered with distinguishing similar devices which for technical reasons have to have separate thing types. In that way, a generic thing type could be listed for users and a corresponding thing handler would change the thing type immediately to a more concrete one, handing over control to the correct specialized handler.
 
+### Thing Categories
+
+A description about channel categories as well as an overview about which categories exist can be found in our [categories overview](../../concepts/categories.html).
```
Comment (maggu2810) [inline docs/documentation/development/bindings/thing-definition.md]: channel categories? Shouldn't this be "thing categories"?

**Comment by triller-telekom** [comment]
Thanks, it's fixed in: #4201

**Pull request merged**